staging/src/k8s.io/pod-security-admission/policy/visitor.go

```diff
@@ -28,9 +28,15 @@ func visitContainers(podSpec *corev1.PodSpec, visitor ContainerVisitor) {
 	for i := range podSpec.InitContainers {
 		visitor(&podSpec.InitContainers[i])
 	}
-	for i := range podSpec.Containers {
-		visitor(&podSpec.Containers[i])
+}
+
+func visitInitContainers(podSpec *corev1.PodSpec, visitor ContainerVisitor) {
+	for i := range podSpec.InitContainers {
+		visitor(&podSpec.InitContainers[i])
 	}
+}
+
+func visitEphemeralContainers(podSpec *corev1.PodSpec, visitor ContainerVisitor) {
```

Comment:
Rather than splitting this into 3 separate functions, I think it would be better to just plumb the fieldpath into the visitor (see https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/pods/helpers.go#L35 for an example)

Reply:
yeah, we explicitly don't want to have three function calls we have to remember to call everywhere to correctly see each container type

Reply:
`k8s.io/pod-security-admission` can't import `k8s.io/kubernetes`, so I'd just copy the pattern to a local function.

Reply:
Is this acceptable to pass the field path every time even if adding details to the error list is disabled ?

Reply:
we can pass a nil value for the fieldpath variable if error details are not enabled